proxy/src/main/java/com/wavefront/agent/AbstractAgent.java

```diff
@@ -701,6 +706,7 @@
   protected ResourceBundle props;
   protected final AtomicLong bufferSpaceLeft = new AtomicLong();
   protected List<String> customSourceTags = new ArrayList<>();
+  protected final Set<String> traceDerivedRedMetricsCustomTagKeys = new HashSet<>();
```

Comment:
use shorter names if possible :)
traceDerivedCustomTagKeys
instead of
traceDerivedRedMetricsCustomTagKeys